src/status_im/ui/screens/discover/views.cljs

```diff
@@ -1,95 +1,114 @@
+                :font  :medium}
+          (str " #" (name tag))]]]]
+      [view styles/tag-count-container
+       [text {:style styles/tag-count
+              :font  :default}
+        (:total discoveries)]]]
+     [components/discover-list-item {:message         (first (:discoveries discoveries))
+                                     :show-separator? false
+                                     :current-account current-account}]]))
+
+(defview popular-hashtags-preview [{:keys [contacts current-account]}]
   (letsubs [popular-tags [:get-popular-tags 10]]
-    [view st/popular-container
-     [title :t/popular-tags false]
+    [view styles/popular-container
+     [components/title :t/popular-tags false :t/all #(do (dispatch [:set :discover-search-tags (map :name popular-tags)])
+                                                         (dispatch [:navigate-to :discover-all-hashtags]))]
      (if (pos? (count popular-tags))
-       [carousel {:pageStyle st/carousel-page-style
-                  :gap       8
-                  :sneak     16
-                  :count     (count popular-tags)}
+       [carousel/carousel {:pageStyle styles/carousel-page-style
+                           :gap       8
+                           :sneak     16
+                           :count     (count popular-tags)}
         (for [{:keys [name]} popular-tags]
-          [discover-popular-list {:tag             name
-                                  :contacts        contacts
-                                  :current-account current-account}])]
-       [text (label :t/none)])]))
+          [top-status-for-popular-hashtag {:tag             name
+                                           :contacts        contacts
+                                           :current-account current-account}])]
+       [text (i18n/label :t/none)])]))
+
+
+(defn empty-discoveries []
+  [view contacts-st/empty-contact-groups
+   ;; todo change the icon
+   [vi/icon :icons/group-big {:style contacts-st/empty-contacts-icon}]
+   [text {:style contacts-st/empty-contacts-text}
+    (i18n/label :t/no-statuses-discovered)]])
 
-(defview discover-recent [{:keys [current-account]}]
-  (letsubs [discoveries [:get-recent-discoveries]]
-    (when (seq discoveries)
-      [view st/recent-container
-       [title :t/recent true]
-       [view st/recent-list
-        (let [discoveries (map-indexed vector discoveries)]
-          (for [[i {:keys [message-id] :as message}] discoveries]
-            ^{:key (str "message-recent-" message-id)}
-            [discover-list-item {:message         message
-                                 :show-separator? (not= (inc i) (count discoveries))
-                                 :current-account current-account}]))]])))
+(defn recent-statuses-preview [current-account discoveries]
+  [view styles/recent-statuses-preview-container
+   [components/title :t/recent false :t/all #(dispatch [:navigate-to :discover-all-recent])]
+   (if (pos? (count discoveries))
```

Comment:
count will always be pos if it exists and is non-empty, more idiomatic would be `(if (seq discoveries))` I think

Reply:
Agree, we'll probably see this a few times, it was a copy paste.